tfhe-zk-pok/src/proofs/pke_v2.rs

```diff
@@ -0,0 +1,2194 @@
+    t: u64,
+    bound: Bound,
+) -> (usize, usize, u64, u64, usize) {
+    let B_r = d as u64 / 2 + 1;
+    let B_bound = {
+        let B = B as f64;
+        let d = d as f64;
+        let k = k as f64;
+
+        (match bound {
+            Bound::GHL => 9.75,
+            Bound::CS => f64::sqrt(2.0 * (d + k) + 4.0),
+        }) * f64::sqrt(sqr(B) + (sqr(d + 2.0) * (d + k)) / 4.0)
+    }
+    .ceil() as u64;
+
+    let m_bound = 2 + B_bound.ilog2() as usize;
+
+    let D = d + k * t.ilog2() as usize;
+    let n = D + 128 * m_bound;
+
+    (n, D, B_r, B_bound, m_bound)
+}
+
+pub fn crs_gen_ghl<G: Curve>(
+    d: usize,
+    k: usize,
+    B: u64,
+    q: u64,
+    t: u64,
+    rng: &mut dyn RngCore,
+) -> PublicParams<G> {
+    let alpha = G::Zp::rand(rng);
+    let B = B * (isqrt((d + k) as _) as u64 + 1);
+    let (n, D, B_r, B_bound, m_bound) = compute_crs_params(d, k, B, q, t, Bound::GHL);
+    PublicParams {
+        g_lists: GroupElements::<G>::new(n, alpha),
+        D,
+        n,
+        d,
+        k,
+        B,
+        B_r,
+        B_bound,
+        m_bound,
+        q,
+        t,
+        hash: core::array::from_fn(|_| rng.gen()),
+        hash_R: core::array::from_fn(|_| rng.gen()),
+        hash_t: core::array::from_fn(|_| rng.gen()),
+        hash_w: core::array::from_fn(|_| rng.gen()),
+        hash_agg: core::array::from_fn(|_| rng.gen()),
+        hash_lmap: core::array::from_fn(|_| rng.gen()),
+        hash_phi: core::array::from_fn(|_| rng.gen()),
+        hash_xi: core::array::from_fn(|_| rng.gen()),
+        hash_z: core::array::from_fn(|_| rng.gen()),
+        hash_chi: core::array::from_fn(|_| rng.gen()),
+    }
+}
+
+pub fn crs_gen_cs<G: Curve>(
+    d: usize,
+    k: usize,
+    B: u64,
+    q: u64,
+    t: u64,
+    rng: &mut dyn RngCore,
+) -> PublicParams<G> {
+    let alpha = G::Zp::rand(rng);
+    let B = B * (isqrt((d + k) as _) as u64 + 1);
+    let (n, D, B_r, B_bound, m_bound) = compute_crs_params(d, k, B, q, t, Bound::CS);
+    PublicParams {
+        g_lists: GroupElements::<G>::new(n, alpha),
+        D,
+        n,
+        d,
+        k,
+        B,
+        B_r,
+        B_bound,
+        m_bound,
+        q,
+        t,
+        hash: core::array::from_fn(|_| rng.gen()),
+        hash_R: core::array::from_fn(|_| rng.gen()),
+        hash_t: core::array::from_fn(|_| rng.gen()),
+        hash_w: core::array::from_fn(|_| rng.gen()),
+        hash_agg: core::array::from_fn(|_| rng.gen()),
+        hash_lmap: core::array::from_fn(|_| rng.gen()),
+        hash_phi: core::array::from_fn(|_| rng.gen()),
+        hash_xi: core::array::from_fn(|_| rng.gen()),
+        hash_z: core::array::from_fn(|_| rng.gen()),
+        hash_chi: core::array::from_fn(|_| rng.gen()),
+    }
+}
+
+pub fn crs_gen<G: Curve>(
+    d: usize,
+    k: usize,
+    B: u64,
+    q: u64,
+    t: u64,
+    rng: &mut dyn RngCore,
+) -> PublicParams<G> {
+    crs_gen_cs(d, k, B, q, t, rng)
```

Comment:
what's the cs vs. ghl notation ? It seems the CRS we use in the test is the ghl one ? @sarah-ek